qemu/tests/ovs_quit.py

```diff
@@ -29,28 +28,31 @@ def get_ovs_ports(ovs):
     utils_path.find_command("ovs-vsctl")
     netdst = params.get("netdst")
     if netdst not in process.system_output("ovs-vsctl list-br"):
-        test.error("%s isn't is openvswith bridge" % netdst)
+        test.cancel("%s isn't an openvswith bridge" % netdst)
 
     deps_dir = data_dir.get_deps_dir("ovs")
-    nic_script = utils_misc.get_path(deps_dir, params["nic_script"])
-    nic_downscript = utils_misc.get_path(deps_dir, params["nic_downscript"])
-    params["nic_script"] = nic_script
-    params["nic_downscript"] = nic_downscript
 
     params["qemu_command_prefix"] = "export SHELL=/usr/bin/bash;"
     params["start_vm"] = "yes"
     params["nettype"] = "bridge"
-    params["nic_model"] = "virtio-net-pci"
+    params["nic_model"] = "virtio-pci"
 
     try:
         ports = get_ovs_ports(netdst)
         env_process.preprocess_vm(test, params, env, params["main_vm"])
         env.get_vm(params["main_vm"])
-    except virt_vm.VMStartError:
-        ports = get_ovs_ports(netdst) - ports
-        if ports:
-            for p in ports:
-                process.system("ovs-vsctl del-if %s %s" % (netdst, p))
-            test.fail("%s not delete after qemu quit." % ports)
+    except (virt_vm.VMCreateError, virt_vm.VMStartError) as err_msg:
+        match_error = "Parameter 'driver' expects"
+        output = getattr(err_msg, 'reason', getattr(err_msg, 'output', ''))
+        if match_error in output:
+            ports = get_ovs_ports(netdst) - ports
+            if ports:
+                for p in ports:
+                    process.system("ovs-vsctl del-if %s %s" % (netdst, p))
+                test.fail("%s not delete after qemu quit." % ports)
+        else:
+            test.fail("VM create failed with not expected error: %s!" % output)
     else:
+        env.get_vm(params["main_vm"]).graceful_shutdown()
+        process.system_output("ovs-vsctl list-br")
```

Comment:
> 1. boot a guest with invalid parameter, the guest should not be started correctly
> 2. ovs-vsctl check, tap device should not be in the ovs bridge

> so, the guest could not boot up is expected. 
> but then, we should use ovs-vsctl to check whether tap device in ovs bridge. Not failed the case

When should the step2 be performed ? Is it only after guest actually boot up with invalid parameters or right after qemu quits?